assets/js/components/Dashboard/DatePicker.tsx

```diff
@@ -0,0 +1,78 @@
+import moment from "moment";
+import React, { ComponentType, useRef, useState } from "react";
+import { Form, Overlay, Popover } from "react-bootstrap";
+import Calendar from "react-calendar";
+
+interface DatePickerProps {
+  selectedDate: string;
+  onChange: (date: string) => void;
+  minDateString?: string;
+  maxDateString?: string;
+}
+
+const DatePicker: ComponentType<DatePickerProps> = ({
+  selectedDate,
+  onChange,
+  minDateString,
+  maxDateString,
+}: DatePickerProps) => {
+  const [inputString, setInputString] = useState(selectedDate);
+  const [showOverlay, setShowOverlay] = useState(false);
+  const ref = useRef(null);
+  const minDate = minDateString ? new Date(minDateString) : undefined;
+  const maxDate = maxDateString ? new Date(maxDateString) : undefined;
+
+  return (
+    <>
+      <Form
+        ref={ref}
+        className="date-picker"
+        onSubmit={(event) => {
+          event.preventDefault();
+          onChange(inputString);
+        }}
+        onClick={() => setShowOverlay(!showOverlay)}
+      >
+        <Form.Control
+          value={inputString}
+          onChange={(input) => {
+            const inputValue = input.target.value;
+            setInputString(inputValue);
+            onChange(inputValue);
+          }}
+        />
+      </Form>
+      <Overlay
+        rootClose
+        onHide={() => setShowOverlay(false)}
+        target={ref.current}
+        placement="bottom-start"
+        show={showOverlay}
+      >
+        {({
+          placement: _placement,
+          arrowProps: _arrowProps,
+          show: _show,
+          popper: _popper,
+          hasDoneInitialMeasure: _hasDoneInitialMeasure,
+          ...props
+        }) => (
+          <Popover className="calendar-popover" {...props}>
```

Comment:
**note:** lodash has both [`_.pick`](https://lodash.com/docs/4.17.15#pick) and [`_.omit`](https://lodash.com/docs/4.17.15#omit) that might make this operation more ergonomic.